### ISSUE TITLE
Menu a11y error

### ISSUE THREAD

**Issue by marksreeves** [BorderTech/wcomponents#358]
We have a WAI-ARIA error in the rendering and control of `WSubMenu` in all `WMenu` types except WMenu.Type.TREE.

Each sub-menu has a wrapper which has role `menuitem[radio|checkbox]?` in menus and `treeitem` in trees. This then contains a control for changing the state of the sub-menu visibility and the content of the submenu. This content has role `menu` in menus and `group` in trees.

The problem is that we apply `aria-expanded` to the wrapper to indicate the sub-menu's state. This is applicable to role `treeitem` but not to `menuitem*`.

In the case of WMenus which are menus we should be applying the aria-expanded attribute to the sub-menu's content wrapper which has role `menu`. We cannot do this in trees because it is not applicable to role `group`.

We have two options I can see (@ricksbrown I would like your opinion on these):

1. Fix the current WMenu code and split the expansion code out for trees and others. This has implications for determining the correct rendering of aWSubMenu's content if the WSubMenu mode is not CLIENT as we will not have the WMenu.Type available to us at the time we render the HTML. This can be overcome in JavaScript but is a royal pain (we are already doing this for a pile of other attributes).
2. We fix the issue for menus but break Type.TREE until we build the new Tree component as part of  #263. Splitting out the schema of trees will free up a lot of other ajax-related issues in the menu space.

The _unfortunate_ side effect of fixing the placement of the aria-expanded attribute is that we will have to base a pile of menu styling and functionality on a classname rather than a WAI-ARIA attribute. I will give this much thought to try to side-step this issue (maybe tie the triggering and style to the role in conjunction with the classname).

**Comment by marksreeves** [comment]
One more problem with a11y in WMenu except Type.TREE when a WSubMenu is mode LAZY or DYNAMIC its wrapper (role menu) does not contain any required role descendants.

**Comment by marksreeves** [comment]
Note to @ricksbrown I took option 1 and fixed it properly. Gives me more incentive to get WTree right!
